### DISCUSSION TITLE
Input image normalization

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#375] [Deep Learning]
> _Archived from the Nengo Discourse forum._ Originally posted by **abhi_kumar** on 2017-05-21. (Forum topic #315, post #1 / id 841)

For most of the deep learning papers it is mentioned to normalize the image. To classify them broadly either images are mapped to the set [0.0, 1.0],  or [-0.5, 0.5]. It would be great if someone could explain the motive behind choosing the set of mapping?.

Regards,
Abhishek

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **Seanny123** on 2017-05-22. (Forum topic #315, post #2 / id 843)

Where does it say to do this? One reason I can imagine is the input domain of the `tanh` or `sigmoid` neurons being limited to `[-1, 1]`, while the `relu` neurons have an input domain of $x>0.0$, but without the specific papers you're referencing, I can't be sure. [This Q&A on stats.stackexchange.com](https://stats.stackexchange.com/q/191762/29949) seems to imply that reducing the covariance by standardizing the inputs is the general motivation.

Further reading:

- [Why do we normalize images by subtracting the dataset's image mean and not the current image mean in deep learning?](https://stats.stackexchange.com/q/211436/29949)
- [How to prepare images for neural network?](https://datascience.stackexchange.com/q/5224/26686)

**Reply by cnrgbot** [reply]
> _Archived from the Nengo Discourse forum._ Originally posted by **abhi_kumar** on 2017-05-22. (Forum topic #315, post #3 / id 846)

Thank you for your answer. When I mentioned `[-0.5, 0.5]` mapping and combined ReLU activation I was talking with respect to [this paper]( http://www.fit.vutbr.cz/~ihradis/pubs.php?file=%2Fpub%2F10922%2Fhradis15CNNdeblurring.pdf&id=10922) which discusses deblurring text using CNN.

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **drasmuss** on 2017-05-22. (Forum topic #315, post #4 / id 859)

Input normalization, or input preprocessing in general, is a big topic.  But, to summarize at a very high level, the idea is that you want to standardize the inputs to your network as much as possible, so that a) learning is more stable (by reducing variability across the training data), b) your network will generalize better to novel data (because the normalization reduces the variability between your training and test data), and c) the inputs will fall within the useful range of your nonlinearities (as Sean pointed out wrt `tanh` or `sigmoid`).

Nowadays input normalization is less critical than it used to be though, due to the prevalence of `relu`'s (which are less sensitive to the range of their inputs) and batch normalization (which accomplishes a similar thing to input normalization but on a layer-by-layer basis).
